examples/fluids/qfunctions/densitycurrent.h

```diff
@@ -185,17 +159,19 @@ CEED_QFUNCTION(ICsDC_Cons)(void *ctx, CeedInt Q,
   for (CeedInt i=0; i<Q; i++) {
     const CeedScalar x[] = {X[0][i], X[1][i], X[2][i]};
     State s = Exact_DC(3, 0., x, 5, ctx);
-    CeedScalar q[5] = {0.};
-    q[0] = s.U.density;
-    for (CeedInt j=0; j<3; j++)
-      q[j+1] = s.U.momentum[j];
-    const CeedScalar Pi = 1. + g*g*(exp(-N*N*x[2]/g) - 1.) / (cp*theta0*N*N);
-    // Multiply internal energy by Exner pressure (Pi)
-    q[4] = s.U.density * (cv*s.Y.temperature*Pi + g*x[2]);
-
-    for (CeedInt j=0; j<5; j++)
-      q0[j][i] = q[j];
-
+    if (gas->primitive) {
+      q0[0][i] = s.Y.pressure;
+      for (CeedInt j=0; j<3; j++)
+        q0[j+1][i] = s.Y.velocity[j];
+      q0[4][i] = s.Y.temperature;
+    } else {
+      q0[0][i] = s.U.density;
+      for (CeedInt j=0; j<3; j++)
+        q0[j+1][i] = s.U.momentum[j];
+      const CeedScalar Pi = 1. + g*g*(exp(-N*N*x[2]/g) - 1.) / (cp*theta0*N*N);
+      // Multiply internal energy by Exner pressure (Pi)
+      q0[4][i] = s.U.density * (cv*s.Y.temperature*Pi + g*x[2]);
```

Comment:
I believe the `theta` defined earlier is the "potential temperature".  The absolute temperature is the product with the Exner function $T = \pi\theta$. But `s.Y.temperature` is meant to be absolute temperature. Ideally we can just use `s.U.E_total` on this line.

Reply:
Alright, just fixed it. Needed to update pressure as well. It does pass locally now but I don't know why I don't see the failures on the CI!

Reply:
Hmm, I manually started a pipeline here. https://gitlab.com/libceed/libCEED/-/pipelines/590481550